src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs

```diff
@@ -104,6 +105,52 @@ public static (string analyzerId, VersionStamp version) GetAnalyzerIdAndVersion(
         public static string GetAnalyzerAssemblyName(this DiagnosticAnalyzer analyzer)
             => analyzer.GetType().Assembly.GetName().Name;
 
+        public static async ValueTask<OptionSet> GetAnalyzerOptionSetAsync(this AnalyzerOptions analyzerOptions, SyntaxTree syntaxTree, CancellationToken cancellationToken)
+        {
+            var configOptions = analyzerOptions.AnalyzerConfigOptionsProvider.GetOptions(syntaxTree);
+            var optionSet = await GetDocumentOptionSetAsync(analyzerOptions, syntaxTree, cancellationToken).ConfigureAwait(false);
+
+            return new AnalyzerConfigOptionSet(configOptions, optionSet);
+        }
+
+        public static T GetOption<T>(this AnalyzerOptions analyzerOptions, Option<T> option, SyntaxTree syntaxTree, CancellationToken cancellationToken)
+        {
+            return GetOptionAsync(analyzerOptions, option, syntaxTree, cancellationToken).GetAwaiter().GetResult();
+        }
+
+        public static async ValueTask<T> GetOptionAsync<T>(this AnalyzerOptions analyzerOptions, Option<T> option, SyntaxTree syntaxTree, CancellationToken cancellationToken)
+        {
+            var configOptions = analyzerOptions.AnalyzerConfigOptionsProvider.GetOptions(syntaxTree);
+            if (configOptions.TryGetOption(option, out var value))
+            {
+                return value;
+            }
+
+            var optionSet = await analyzerOptions.GetDocumentOptionSetAsync(syntaxTree, cancellationToken).ConfigureAwait(false);
+            return optionSet is object
+                ? optionSet.GetOption(option)
+                : option.DefaultValue;
+        }
+
+        public static T GetOption<T>(this AnalyzerOptions analyzerOptions, PerLanguageOption<T> option, string language, SyntaxTree syntaxTree, CancellationToken cancellationToken)
+        {
+            return GetOptionAsync(analyzerOptions, option, language, syntaxTree, cancellationToken).GetAwaiter().GetResult();
+        }
+
+        public static async ValueTask<T> GetOptionAsync<T>(this AnalyzerOptions analyzerOptions, PerLanguageOption<T> option, string language, SyntaxTree syntaxTree, CancellationToken cancellationToken)
```

Comment:
Can't this and the the GetOptionAsync<T>() that takes an `Option<T>` just call into a shared implementation that takes an OptionKey?